### DISCUSSION TITLE
Cannot find namespace 'p5' in [@types/p5]

### DISCUSSION THREAD

**Discussion by lukeellison** [DefinitelyTyped/DefinitelyTyped#63302] [Issues with a @types package]
When creating a sketch in my simple setup with typescript I get the error "Cannot find namespace 'p5'" when declaring something of type `p5.Element` or `p5.Vector`.
The warning is suppressed when you add the line `import * as p5 from "p5"` but if you're working in an environment without imports that doesn't work.
The solution I have so far is to add
```
declare namespace p5 {
  interface Element {
    addClass(className: string): void;
    removeClass(className: string): void;
    style(property: string, value: string): void;
    position(x: number, y: number): void;
    mouseReleased(callback: () => void): void;
    html(html: string): void;
    value(): number;
  }
}
```
But this is only for the functions I'm using and I have to add a new one each time I use a new function. Also it does not include the docstrings.
It would be better to add these to the library if possible? Unless perhaps I have something wrong with my setup and this is avoidable.

Thanks!

[a comment by an automated account is not shown]

**Comment by Zalastax** [comment]
Sorry, but I no longer maintain p5 types.
